servers/shares-api/schema.graphql

```diff
@@ -1,3 +1,77 @@
-type PocketShare {
-  id: ID!
-}
\ No newline at end of file
+extend schema
+  @link(url: "https://specs.apollo.dev/link/v1.0")
+  @link(
+    url: "https://specs.apollo.dev/federation/v2.3"
+    import: ["@key", "@composeDirective", "@tag"]
+  )
+  # The link directive is required to federate @constraint
+  # It doesn't actually have to be a real spec, but it would be good
+  # to write one and replace this.
+  @link(url: "https://myspecs.dev/myDirective/v1.0", import: ["@constraint"])
+  @composeDirective(name: "@constraint")
+
+scalar Url
+scalar Markdown
+scalar ISOString
+
+type PocketShare @key(fields: "slug targetUrl") {
+  slug: ID!
+  shareUrl: Url!
+  targetUrl: Url!
+  createdAt: ISOString
+  context: ShareContext
+}
+
+type ShareContext {
+  """
+  A user-provided comment/note on the shared content.
+  """
+  note: Markdown
+  """User-provided highlights of the content"""
+  highlights: [ShareHighlight!]
+}
+
+type ShareHighlight {
```

Comment:
wondering: how is this expected to render on the client?

Reply:
I think like a list of quotes (which is why I didn't include the article reconstruction context)

See upnext for prior art. I didn't add notes on the ShareHighlgiht yet to reduce complexity but we could

![image](https://github.com/Pocket/pocket-monorepo/assets/34227334/04bde205-b4e2-4913-8082-7608175750cc)

Reply:
thanks for clarifying @kschelonka !